ext/ffi/00_ffi.js

```diff
@@ -27,9 +33,26 @@ const {
   SafeArrayIterator,
 } = primordials;
 
+/**
+ * @param {BufferSource} source
+ * @returns {number}
+ */
+function getBufferSourceByteLength(source) {
+  if (ArrayBufferIsView(source)) {
+    if (TypedArrayPrototypeGetSymbolToStringTag(source) !== undefined) {
+      // TypedArray
+      return TypedArrayPrototypeGetByteLength(source);
+    } else {
+      // DataView
+      return DataViewPrototypeGetByteLength(source);
```

Comment:
DataView is actually not supported by the Rust side but it doesn't really hurt here either.